### PR TITLE
Migrate volume and loop unit tests to component.yaml and verify with v2

### PR DESCRIPTION
**Pull request by Tomcli**
**Which issue is resolved by this Pull Request:** 
Related #670 

**Description of your changes:**
Migrate volume and loop unit tests (alphabetical order from 'v' to the end) to use the new component.yaml that also works with v2. The loop dsl currently only works on single parallel loop for IR, but not for nested loop such as parallel loop inside another parallel loop. 

The volumeSnapshotOp is no longer relevant because it was producing the volumeSnapshot alpha API which only can work with Kubernetes 1.17 or below. In the latest version (KFP 1.7.0+) we need to use Kubernetes 1.18 or above in order to run Argo and Tekton CRDs.

**Environment tested:**

* Python Version (use `python --version`):
* Tekton Version (use `tkn version`):
* Kubernetes Version (use `kubectl version`):
* OS (e.g. from `/etc/os-release`):

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

### CODE REVIEW

File: sdk/python/tests/compiler/testdata/withparam_global.py [hunk 1]
```diff
@@ -23,32 +24,63 @@ def empty(self):
...
+outputs:
+  - {name: out, type: String}
+implementation:
+    container:
+        image: python:alpine3.6
+        command: ['sh', '-c']
+        args:
+        - |
+          set -e
+          python -c "import json; import sys; json.dump([i for i in range(20, 31)], open(\'\"$0\"\', 'w'))"
```
Comment (yhwang) [inline sdk/python/tests/compiler/testdata/withparam_global.py:38]: `open(\'\"$0\"\', 'w'))` should be
`open('$0', 'w'))`

File: sdk/python/tests/compiler/testdata/withparam_global_dict.py [hunk 2]
```diff
@@ -23,31 +24,65 @@ def empty(self):
...
+outputs:
+  - {name: out, type: String}
+implementation:
+    container:
+        image: python:alpine3.6
+        command: ['sh', '-c']
+        args:
+        - |
+          set -e
+          python -c "import json; import sys; json.dump([i for i in range(20, 31)], open(\'\"$0\"\', 'w'))"
```
Comment (yhwang) [inline sdk/python/tests/compiler/testdata/withparam_global_dict.py:38]: same here, `open('$0', 'w'))`

File: sdk/python/tests/compiler/testdata/withparam_output.py [hunk 3]
```diff
@@ -23,32 +24,63 @@ def empty(self):
...
+outputs:
+  - {name: out, type: String}
+implementation:
+    container:
+        image: python:alpine3.6
+        command: ['sh', '-c']
+        args:
+        - |
+          set -e
+          python -c "import json; import sys; json.dump([i for i in range(20, 31)], open(\'\"$0\"\', 'w'))"
```
Comment (yhwang) [inline sdk/python/tests/compiler/testdata/withparam_output.py:38]: same here: `open('$0', 'w')`

File: sdk/python/tests/compiler/testdata/withparam_output_dict.py [hunk 4]
```diff
@@ -23,33 +24,63 @@ def empty(self):
...
+outputs:
+  - {name: out, type: String}
+implementation:
+    container:
+        image: python:alpine3.6
+        command: ['sh', '-c']
+        args:
+        - |
+          set -e
+          python -c "import json; import sys; json.dump([{\'a\': 1, \'b\': 2}, {\'a\': 10, \'b\': 20}], open(\'\"$0\"\', 'w'))"
```
Comment (yhwang) [inline sdk/python/tests/compiler/testdata/withparam_output_dict.py:38]: `open(\'\"$0\"\', 'w')`==> `open('$0', 'w')`

File: sdk/python/tests/compiler/testdata/withparam_global.py (hunk 1 above)
Comment (Tomcli) [reply sdk/python/tests/compiler/testdata/withparam_global.py]: done

**Review by yhwang — APPROVED** [review]
/lgtm

**Pull request merged**